Automattic/pocket-casts-ios
# Add new Medium "Now Playing" widget

Follow up to https://github.com/Automattic/pocket-casts-ios/pull/1343 that introduces the new medium Now Playing widget from Lkt7fuf9Nq3XvfAFPCfpTD-fi-995_4119

Also adapts the existing "nothing playing" view for the medium size.

**iOS 17**
| light | dark |
| -- | -- |
| ![Simulator Screenshot - iPhone 15 Plus - 2024-07-04 at 14 54 43](https://github.com/Automattic/pocket-casts-ios/assets/4081020/7f7bd20e-0633-4995-ae2b-91621358b645) | ![Simulator Screenshot - iPhone 15 Plus - 2024-07-04 at 14 55 18](https://github.com/Automattic/pocket-casts-ios/assets/4081020/aa3916dc-e894-44c8-8dfc-be08c8ba3d85) |


**iOS 16**
| light | dark |
| -- | -- |
| ![Simulator Screenshot - iPhone SE (3rd generation) - 2024-07-04 at 15 42 59](https://github.com/Automattic/pocket-casts-ios/assets/4081020/120229dd-7912-47c8-aa15-6cfcf27f7500) | ![Simulator Screenshot - iPhone SE (3rd generation) - 2024-07-04 at 15 42 35](https://github.com/Automattic/pocket-casts-ios/assets/4081020/caca2864-4172-4447-8643-b40ee6e580a3) |

**Nothing Playing**
| light | dark |
| -- | -- |
| ![Simulator Screenshot - iPhone 15 Plus - 2024-08-01 at 16 23 14](https://github.com/user-attachments/assets/12bbb822-6513-4435-ad41-af8af973f0bf) | ![Simulator Screenshot - iPhone 15 Plus - 2024-08-01 at 16 23 00](https://github.com/user-attachments/assets/c4b61812-0922-4ca8-b149-88104c68319c) |


## To test

1. pull branch, build and run on device or simulator
2. Open the widget chooser, scroll through the options for Pocket Casts. You should see the new medium Now Playing widget in both styles after the "small" variation
3. Add one (or both) to your home screen
4. The widget should match the designs
5. If you added the white version, flipping into "dark" mode should turn it black. Or if it was already black, flipping into "light" mode should turn it white.
6. In iOS 17, the widget should be interactive. Tapping play should start the podcast, tapping pause should stop it.
   - tapping anywhere else should open Pocket Casts to the episode
8. In iOS 16, tapping anywhere should open Pocket Casts to the episode
9. When there isn't a "Now Playing" episode, the default state will match the "small" size and simply show the current Pocket Casts icon -- Waiting on design to review this since it won't work well as-is if a custom icon is chosen p1720121313396209-slack-C05RR9P9RAT

NOTE: I'm not entirely sure when `showsWidgetContainerBackground` will ever be false for these widgets, but if you want to simulate that you can set `Line 8` of `NowPlayingEntryView.swift` force `var showsWidgetBackground = false` and then see how it renders. It won't be pretty because padding is removed for some reason, but that isn't done as part of this PR.


## Checklist
